Revert:
2015-02-05  H.J. Lu  <hongjiu.lu@intel.com>

	* lto-plugin.c (claim_file_handler): Pass handle to
	release_input_file.

2015-02-05  H.J. Lu  <hongjiu.lu@intel.com>

	* lto-plugin.c (claim_file_handler): Call release_input_file only
	if file is claimed.

--- lto-plugin/lto-plugin.c	(revision 220461)
+++ lto-plugin/lto-plugin.c	(revision 220454)
@@ -998,9 +998,6 @@ claim_file_handler (const struct ld_plug
 
   *claimed = 1;
 
-  if (release_input_file)
-    release_input_file (file->handle);
-
   goto cleanup;
 
  err:
@@ -1010,6 +1007,9 @@ claim_file_handler (const struct ld_plug
   if (obj.objfile)
     simple_object_release_read (obj.objfile);
 
+  if (release_input_file)
+    release_input_file (file);
+
   return LDPS_OK;
 }
 
